File: DIP/Python/Relationships/WithViolation/research.py
from relationship import Relationship
from relationships import Relationships


class Research:
    def __init__(self, relationsships: Relationships) -> None:
        relations = relationsships.relations
        for relation in relations:
            if relation[0].name == "Jefté" and relation[1] == Relationship.PARENT:
                print(f"Jefté has a child called {relation[2].name}")

File: DIP/JavaScript/WithoutViolation/research.js
const { Relationship } = require("./relationship");

class Research {
  // constructor(relationships) {
  //   let relations = relationships.data;
  //   for (let rel of relations.filter(
  //     (r) => r.from.name === "Jefté" && r.type == Relationship.parent
  //   )) {
  //     console.log(`Jefté has a child named ${rel.to.name}`);
  //   }
  // }

  constructor(browser) {
    for (let p of browser.findAllChildrenOf("Jefté")) {
      console.log(`Jefté has a child named ${p.name}`);
    }
  }
}

module.exports.Research = Research;
